python/python/tests/tests_server_modules/test_ft.py

```diff
@@ -303,3 +318,408 @@ async def _create_test_index_for_ft_explain_commands(
             )
             == OK
         )
+
+    @pytest.mark.parametrize("cluster_mode", [True])
+    @pytest.mark.parametrize("protocol", [ProtocolVersion.RESP2, ProtocolVersion.RESP3])
+    async def test_ft_aggregate_with_bicycles_data(
+        self, glide_client: GlideClusterClient, protocol
+    ):
+        prefixBicycles = "{bicycles}:"
+        indexBicycles = prefixBicycles + str(uuid.uuid4())
+        await TestFt._create_index_for_ft_aggregate_with_bicycles_data(
+            self=self,
+            glide_client=glide_client,
+            index_name=indexBicycles,
+            prefix=prefixBicycles,
+        )
+        await TestFt._create_json_keys_for_ft_aggregate_with_bicycles_data(
+            self=self, glide_client=glide_client, prefix=prefixBicycles
+        )
+        time.sleep(self.sleep_wait_time)
+
+        # Run FT.AGGREGATE command with the following arguments: ['FT.AGGREGATE', '{bicycles}:1e15faab-a870-488e-b6cd-f2b76c6916a3', '*', 'LOAD', '1', '__key', 'GROUPBY', '1', '@condition', 'REDUCE', 'COUNT', '0', 'AS', 'bicycles']
+        result = await ft.aggregate(
+            glide_client,
+            indexName=indexBicycles,
+            query="*",
+            options=FtAggregateOptions(
+                loadFields=["__key"],
+                clauses=[GroupBy(["@condition"], [Reducer("COUNT", [], "bicycles")])],
+            ),
+        )
+        assert await ft.dropindex(glide_client, indexName=indexBicycles) == OK
+        if protocol == ProtocolVersion.RESP2:
+            for e in result:
+                if e.get(b"condition") == b"new":
+                    assert e.get(b"bicycles") == b"5"
+                elif e.get(b"condition") == b"refurbished":
+                    assert e.get(b"bicycles") == b"1"
+                elif e.get(b"condition") == b"used":
+                    assert e.get(b"bicycles") == b"4"
+        elif protocol == ProtocolVersion.RESP3:
+            for e in result:
+                if e.get(b"condition") == b"new":
+                    assert cast(float, e.get(b"bicycles")) == 5.0
+                elif e.get(b"condition") == b"refurbished":
+                    assert cast(float, e.get(b"bicycles")) == 1.0
+                elif e.get(b"condition") == b"used":
+                    assert cast(float, e.get(b"bicycles")) == 4.0
```

Comment:
Create 2 expected and make a single assert
or even
```
expected = [{b'condition': is_resp2 ? b'5' : 5.0, ....
```

Reply:
Updated.